starlite/contrib/htmx/utils.py

```diff
@@ -0,0 +1,28 @@
+from enum import Enum
+
+HTMX_STOP_POLLING = 286
+
+
+class HX(str, Enum):

```

Comment:
HX is not a good name. Please rename it to be descriptive, e.g. `HtmxHeaders` etc.

Reply:
I'd like to point out that, I haven't implemented encode and decode_json as it returns bytes and htmx cannot understand headers in bytes . For the time being, I am using json dumps.

There are two features in my mind that I'd like to share:
```
1)  Template response with ability to push_url, reswap, retarget and trigger_event.

class HtmxTemplateResponse(Template):
    def __init__(self, push: str, 
        reswap: str, 
        retarget: str, 
        trigger: str, 
        **kwargs) -> None:

```

```
2)  Starlite provides response_headers parameter in http handlers,
 which I would like to use as follows:

@get('/contact-us',
        response_headers=htmx_headers(push='/contact-us')
     )  # htmx_headers returns {'hx-push-url': ResponseHeader(value='/contact-us)}
async def contact(request: HTMXRequest) -> Any:
    ...
    return 'Success!'

- Frees the dev to return any type of response
```

Reply:
> I haven't implemented encode and decode_json as it returns bytes and htmx cannot understand headers in bytes

Decode the bytes then. `encode_json` in particular is important since the stdlib `json` module doesn't support all the types `encode_json` does. Having only *some* types work with HTMX is suboptimal and unintuitive.

Reply:
Ok.